colossalai/inference/tensor_parallel/modeling/chatglm2.py

```diff
@@ -466,19 +465,18 @@ def chatglm_flash_attn_kvcache_forward(
         value_layer = value_layer.reshape(
             -1, self.num_multi_query_groups_per_partition, self.hidden_size_per_attention_head
         )
+
         if infer_state.is_context_stage:
             # first token generation:
             # copy key and value calculated in current step to memory manager
-
             copy_kv_to_mem_cache(
                 infer_state.decode_layer_id,
                 key_layer,
                 value_layer,
                 infer_state.context_mem_index,
                 infer_state.cache_manager,
             )
-
-            attn_output = torch.empty_like(query_layer.view(-1, self.projection_size))
+            attn_output = torch.empty_like(query_layer.contiguous().view(-1, self.projection_size))
```

Comment:
why is that ? why we need it to be contiguous?

Reply:
encountered Runtime error when not using `contiguous()`